tests/test_dulwich.py

```diff
@@ -26,3 +30,21 @@ def test_dulwich_github_compat(mocker: MockerFixture, algorithm: bytes):
     strings = iter((b"ssh-rsa", key_data))
     packet.get_string = lambda: next(strings)
     _process_public_key_ok_gh(auth, None, None, packet)
+
+
+def test_dulwich_get_ref(tmp_dir: TmpDir, scm: Git):
+    backend = DulwichBackend(tmp_dir)
+    tmp_dir.gen("foo", "foo")
+    scm.add_commit("foo", message="foo")
+
+    scm.set_ref("refs/exp/base", "refs/heads/master")
+
+    custom_ref = backend.get_ref("refs/exp/base", follow=False)
+    assert str(custom_ref).rstrip("\n") == "refs/heads/master"
+
+    head = scm.get_rev()
+    tag = "my_tag"
+    scm.tag(["-a", tag, "-m", "Annotated Tag"])
+
+    tag_ref = backend.get_ref(f"refs/tags/{tag}", follow=False)
```

Comment:
Is there a reason this is `follow=False`? Tags aren't symbolic refs, and should always be retrieved with `follow=True` (which implies that `get_ref` should return a resolved object SHA and not the raw content of a [symbolic] ref file)

Reply:
> Is there a reason this is follow=False?

Just because it is how I reached the bug on DVC side:

- `describe`: https://github.com/iterative/dvc/blob/master/dvc/repo/experiments/show.py#L59
- `get_ref` with `follow=False`: https://github.com/iterative/scmrepo/blob/main/scmrepo/git/backend/dulwich/__init__.py#L607